### PR TITLE
Remove RNTester/RNTester-macOS/TurboModuleCxx and download files from microsoft/react-native-windows

### PR DESCRIPTION
**Pull request by ZihanChen-MSFT**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Some files in `RNTester/RNTester-macOS/TurboModuleCxx` are copied from `microsoft/react-native-windows` before.
This change downloads all of them by cocpapods.

I have created 3 cocoapods in `ReactTurboModuleCxx` folder:
- **React-TurboModuleCxx-WinRTShared**: This will be deleted and merged to `*-WinRTPort` after proper files in `react-native-windows` are updated later (very soon)
- **React-TurboModuleCxx-WinRTPort**: A minimum port of WinRT to macOS because source files comming from `react-native-window`s are based on WinRT. Removing the dependency to make them "real" C++ is in the plan.
- **React-TurboModuleCxx-RNW**: Download proper files from `react-native-windows`.

## Changelog

[macOS] [Fixed] - Remove RNTester/RNTester-macOS/TurboModuleCxx and download files from microsoft/react-native-windows

## Test Plan

Manually built

**NOTE**: This change affects the `snapshot/screenshot` test page, but the test page was broken before I made this change. I will address it in another pull request.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-macos/pull/630)

### CODE REVIEW

File: ReactTurboModuleCxx/React-TurboModuleCxx-RNW.podspec [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                             "vnext/Microsoft.ReactNative/TurboModulesProvider.{h,cpp}",
+                             "vnext/Microsoft.ReactNative.Cxx/JSValue*.{h,cpp}",
+                             "vnext/Microsoft.ReactNative.Cxx/Module*.{h,cpp}",
+                             "vnext/Microsoft.ReactNative.Cxx/React*.{h,cpp}",
+                             "vnext/Microsoft.ReactNative.Cxx/StructInfo.h",
+                             "vnext/Microsoft.ReactNative.Cxx/NativeModules.h"
+  s.library                = "stdc++"
+  s.pod_target_xcconfig    = { "USE_HEADERMAP" => "YES",
+                               "CLANG_CXX_LANGUAGE_STANDARD" => "c++17" ,
+                               "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/RCT-Folly\"" "\"$(PODS_ROOT)/Headers/Private/React-callinvoker\"" "\"$(PODS_ROOT)/Headers/Private/ReactCommon\"" }
```
Comment (alloy) [inline ReactTurboModuleCxx/React-TurboModuleCxx-RNW.podspec:32]: Did you really need to have `$(PODS_ROOT)/RCT-Folly` even though you depend on `RCT-Folly` below?

As for the other two, looks like those should perhaps become public instead?

File: ReactTurboModuleCxx/React-TurboModuleCxx-WinRTPort.podspec [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  s.homepage               = "https://github.com/microsoft/react-native-macos"
+  s.license                = package["license"]
+  s.author                 = "Microsoft Corporation"
+  s.platforms              = { :ios => "10.0", :tvos => "10.0", :osx => "10.13" }
+  s.source                 = source
+  s.source_files           = "WinRTPort/*.{h,cpp,mm}"
+  s.library                = "stdc++"
+  s.pod_target_xcconfig    = { "USE_HEADERMAP" => "YES",
+                               "CLANG_CXX_LANGUAGE_STANDARD" => "c++17" ,
+                               "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/Headers/Private/React-callinvoker\"" }
```
Comment (alloy) [inline ReactTurboModuleCxx/React-TurboModuleCxx-WinRTPort.podspec:27]: Same as above.

File: RNTester/Podfile.lock [hunk 3]
```diff
@@ -501,32 +521,35 @@ SPEC CHECKSUMS:
...
+  React-TurboModuleCxx-WinRTShared: b071c053b74a94a204ea2dc87b622795aebd067c
+  ReactCommon: 0b2b124b9b91c9cd5bd7538ba265538951e46fbf
+  Yoga: e79d814734797bc6ff12b2b7702224983a425047
   YogaKit: f782866e155069a2cca2517aafea43200b01fd5a
 
-PODFILE CHECKSUM: f92cb3c61bb71bceed9b9afe3944fbd8451be8b9
+PODFILE CHECKSUM: 453d98c1a4e763cdbf636ce3570660198629bdf1
 
-COCOAPODS: 1.8.4
+COCOAPODS: 1.9.3
```
Comment (alloy) [inline RNTester/Podfile.lock:555]: Can you either install CocoaPods v1.8.4 system-wide `gem install cocoapods --version '= 1.8.4'` _or_ use the provided `Gemfile` to use the correct version?

```
gem install bundler
bundle install
bundle exec pod install
```

**Review by alloy — CHANGES_REQUESTED** [review]
Nice!

Can you at least update the `Podfile.lock` changes by installing the expected CocoaPods version?

File: ReactTurboModuleCxx/React-TurboModuleCxx-RNW.podspec (hunk 1 above)
Comment (ZihanChen-MSFT) [reply ReactTurboModuleCxx/React-TurboModuleCxx-RNW.podspec]: fixed

File: ReactTurboModuleCxx/React-TurboModuleCxx-WinRTPort.podspec (hunk 2 above)
Comment (ZihanChen-MSFT) [reply ReactTurboModuleCxx/React-TurboModuleCxx-WinRTPort.podspec]: fixed

File: RNTester/Podfile.lock (hunk 3 above)
Comment (ZihanChen-MSFT) [reply RNTester/Podfile.lock]: fixed

**Review by alloy — APPROVED** [review]
Great! :shipit:

**Pull request merged**